infrastructure/cdn-in-a-box/cache/Dockerfile

```diff
@@ -51,109 +54,84 @@ RUN dnf -y install epel-release && \
+    rm -f /astats_over_http.c /Makefile.am
 
-# You need to do this because the RPM in the ATS archives is just all kinds of messed-up
-RUN chmod 755 /usr/lib64/trafficserver /etc/trafficserver/body_factory /etc/trafficserver/body_factory/default
-RUN mkdir -p /var/trafficserver /opt/ort && \
-    dd if=/dev/zero bs=1M count=1000 of=/var/trafficserver/cache && \
-    chown -R ats:ats /etc/trafficserver/ /var/trafficserver/ /opt/ort /usr/lib64/trafficserver/ && \
-    sed -i 's/STRING 8080 8080:ipv6/STRING 80 80:ipv6/' /etc/trafficserver/records.config
+# traffic_server needs different ownership than trafficserver-7.1.4-2.el7.x86_64.rpm sets
+RUN chown -R ats:ats /etc/trafficserver/ /usr/lib64/trafficserver/
 
-RUN setcap CAP_NET_BIND_SERVICE=+eip /bin/traffic_server && setcap CAP_NET_BIND_SERVICE=+eip /bin/traffic_manager && setcap CAP_NET_BIND_SERVICE=+eip /bin/trafficserver && setcap CAP_NET_BIND_SERVICE=+eip /bin/traffic_cop
 
 WORKDIR /opt
 
-ADD infrastructure/cdn-in-a-box/ort /opt/ort/
-ADD traffic_control/clients/python /opt/Apache-TrafficControl/
-
-RUN touch /var/log/ort.log && \
-	pip3 install ./Apache-TrafficControl && \
-	pip3 install ./ort && \
-	cp ort/traffic_ops_ort.crontab /etc/cron.d/traffic_ops_ort-cron-template && \
-	cp ort/traffic_ops_ort.logrotate /etc/logrotate.d/ort
+ADD infrastructure/cdn-in-a-box/cache/traffic_ops_ort.crontab /etc/cron.d/traffic_ops_ort-cron-template
+ADD infrastructure/cdn-in-a-box/cache/traffic_ops_ort.logrotate /etc/logrotate.d/ort
 
 ADD infrastructure/cdn-in-a-box/cache/run.sh infrastructure/cdn-in-a-box/traffic_ops/to-access.sh infrastructure/cdn-in-a-box/enroller/server_template.json /
 
 COPY infrastructure/cdn-in-a-box/dns/set-dns.sh \
      infrastructure/cdn-in-a-box/dns/insert-self-into-dns.sh \
+     infrastructure/cdn-in-a-box/cache/uname \
      /usr/local/sbin/
+# Copy systemctl.sh to /usr/bin specifically because t3c runs /bin/systemctl by absolute path
+COPY cache-config/testing/docker/ort_test/systemctl.sh /usr/bin/systemctl
 
 ARG ORT_RPM=infrastructure/cdn-in-a-box/cache/trafficcontrol-cache-config.rpm
 ADD $ORT_RPM /
-RUN rpm -Uvh --nodeps /$(basename $ORT_RPM) &&\
+RUN rpm -Uvh /$(basename $ORT_RPM) &&\
     rm /$(basename $ORT_RPM)
 CMD /run.sh
 
-FROM common-cache-server-layers AS mid
+FROM common-traffic-server-dependencies AS get-delve
+RUN dnf -y install golang && \
+    go get -u github.com/go-delve/delve/cmd/dlv
+
+FROM common-cache-config-layers AS mid
 COPY infrastructure/cdn-in-a-box/mid/init.d/ /opt/init.d/
 
-FROM common-cache-server-layers AS edge
+FROM mid AS mid-debug
+COPY --from=get-delve /root/go/bin /usr/bin
+COPY infrastructure/cdn-in-a-box/cache/init-debug-scripts.sh /opt/init.d/
+
+FROM common-cache-config-layers AS edge
 COPY infrastructure/cdn-in-a-box/edge/init.d/ /opt/init.d/
+
+FROM edge AS edge-debug
+COPY --from=get-delve /root/go/bin /usr/bin
+COPY infrastructure/cdn-in-a-box/cache/init-debug-scripts.sh /opt/init.d/
+
+FROM edge
```

Comment:
what does this hanging `FROM` line do?

Reply:
Makes it so that the image builds using the `edge` target by default instead of the `edge-debug` target if no target is specified. Added a comment explaining why in 5c3def0336.